cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -272,17 +282,30 @@ enum PartialAuthorizationAnswer {
 
 #[serde_as]
 #[derive(Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
+#[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
 struct AuthorizationCall {
+    #[cfg_attr(
+        feature = "wasm",
+        tsify(optional, type = "string|{type: string, id: string}")
+    )]
     principal: Option<JsonValueWithNoDuplicateKeys>,
+    #[cfg_attr(feature = "wasm", tsify(type = "string|{type: string, id: string}"))]
     action: JsonValueWithNoDuplicateKeys,
+    #[cfg_attr(
+        feature = "wasm",
+        tsify(optional, type = "string|{type: string, id: string}")
+    )]
     resource: Option<JsonValueWithNoDuplicateKeys>,
     #[serde_as(as = "MapPreventDuplicates<_, _>")]
+    #[cfg_attr(feature = "wasm", tsify(optional, type = "Record<string, any>"))]
     context: HashMap<String, JsonValueWithNoDuplicateKeys>,
     /// Optional schema in JSON format.
     /// If present, this will inform the parsing: for instance, it will allow
     /// `__entity` and `__extn` escapes to be implicit, and it will error if
     /// attributes have the wrong types (e.g., string instead of integer).
     #[serde(rename = "schema")]
+    #[cfg_attr(feature = "wasm", tsify(optional, type = "Record<string, any>"))]
```

Comment:
We'll wanna come back to this and change it to the schema type when we export that